### PR TITLE
[Rust/Cosmos]: Move Cosmos blockchain into Rust

### PR DESCRIPTION
**Pull request by satoshiotomakan**
## Description

Move all cosmos-sdk based blockchains to Rust.

## Breaking changes

* [Cosmos::Proto::ExecuteContract](https://github.com/trustwallet/wallet-core/blob/4a5fc03108436a9ab674a96c4b98a463d8f56f26/src/proto/Cosmos.proto#L120-L126) has been removed. Consider using [Cosmos::Proto::WasmExecuteContractGeneric](https://github.com/trustwallet/wallet-core/blob/4a5fc03108436a9ab674a96c4b98a463d8f56f26/src/proto/Cosmos.proto#L226-L240) instead.
* Signature has been changed for the [Cosmos::Proto::AuthGrant::StakeAuthorization](https://github.com/trustwallet/wallet-core/blob/4a5fc03108436a9ab674a96c4b98a463d8f56f26/src/proto/Cosmos.proto#L298) transaction message due to the changed Protobuf serialization scheme. Unfortunately, `libprotobuf` and `quick-protobuf` serialize in a different sequences, but the encoded data is still valid and **should** be accepted by the network.

## TODO

- [x] Move Thorchain signing logic to Rust
- [ ] Broadcast `StakeAuthorization` transaction
- [x] Remove C++ implementation
- [ ] Move all C++ tests to Rust (at the next PR)
- [x] Add fuzz tests

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: include/TrustWalletCore/TWBlockchain.h [hunk 1]
```diff
@@ -65,6 +65,8 @@ enum TWBlockchain {
     TWBlockchainSui = 50,
     TWBlockchainGreenfield = 51,
     TWBlockchainInternetComputer = 52,
+    TWBlockchainNativeEvmos = 55, // Cosmos
+    TWBlockchainNativeInjective = 56, // Cosmos
```
Comment (Milerius) [inline include/TrustWalletCore/TWBlockchain.h:69]: how it was working before?

File: registry.json [hunk 2]
```diff
@@ -4365,7 +4366,7 @@
     "coinId": 10000060,
     "symbol": "INJ",
     "decimals": 18,
-    "blockchain": "Cosmos",
+    "blockchain": "NativeInjective",
```
Comment (Milerius) [inline registry.json:4369]: why do we need those change(s)?

File: include/TrustWalletCore/TWBlockchain.h (hunk 1 above)
Comment (satoshiotomakan) [reply include/TrustWalletCore/TWBlockchain.h:69]: Before, we just checked the coin type, and used different public key prefix types for NativeEvmos and NativeInjective coins.
It smells like a bad pattern, so in Rust provide `CosmosContext` where the `CosmosContext::PublicKey` solves this problem.

File: registry.json (hunk 2 above)
Comment (satoshiotomakan) [reply registry.json:4369]: Answered here https://github.com/trustwallet/wallet-core/pull/3498#discussion_r1395713957

**Pull request merged**